crates/orderbook/src/ipfs.rs

```diff
@@ -53,11 +70,40 @@ impl Ipfs {
 
     fn prepare_url(&self, cid: &str) -> Url {
         let mut url = shared::url::join(&self.base, &format!("ipfs/{cid}"));
-        if let Some(query) = &self.query {
-            url.set_query(Some(query.as_str()));
+        if let Some(jwt) = &self.auth_token {
+            url.set_query(Some(&format!("pinataGatewayToken={}", jwt)));
```

Comment:
This subtle change allows the auth token to be used for the private gateway both for fetching and for uploading. The way this was formatted before was bound to the "fetch" endpoint.